scripts/fetch/fetch.precip_shapefiles.R

```diff
@@ -1,9 +1,60 @@
+# Download all available shapefiles in our date range
+fetch.precip_shapefiles <- function(viz = as.viz('precip_shapefiles')){
+  
+  precip_page_urls <- get_precip_page_urls(viz)
+  
+  precip_dir <- viz[["location"]]
+  if(!dir.exists(precip_dir)) dir.create(precip_dir)
+  
+  for(precip_page_url in precip_page_urls) {
+    
+    precip_files <- get_precip_files_on_page(precip_page_url)
+
+    for(dFile in precip_files) {
+      outfile <- file.path(precip_dir, dFile)
+      if(!file.exists(outfile)) {
+        download.file(paste0(precip_page_url, dFile), outfile)
+      }
+    }
+  }
+  # update the timstamp of the last outfile to make sure the directory timestamp
+  # is always updated by now, even if there were no new files to download
+  Sys.setFileTime(precip_dir, Sys.time())
+}
+
+# Monitor the timestamp of the last-available precip file in our date range
 fetchTimestamp.precip_shapefiles <- function(viz) {
-  invisible()
+  
+  # find the latest page of precip data
+  precip_page_urls <- get_precip_page_urls(viz)
+  last_precip_page <- precip_page_urls[length(precip_page_urls)]
+  
+  # find the last zip file on that last page
+  precip_files <- get_precip_files_on_page(last_precip_page)
+  last_precip_file <- sort(precip_files, decreasing=TRUE)[1]
+  
+  # construct the url and pass to urlLastModified, which will create/modify the
+  # timestamp file as needed
```

Comment:
Note that we should be looking at the date stamps of the files. Looks like there's a simple way to read the html into a table with rvest::html_table. We could save the last modified for each file and not get it unless our local copy is stale. I'm not sure how this would integrate with the timeStamp file in vizlab, but it would be a way to have this fetcher use it's old results to know what it needs to fetch.